src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -118,8 +118,16 @@ const useFetchChannelHook = () => {
       const isLocationChannel = channel?.channel_type === 2 || channel?.type_channel === 2;
       const isDeletedMessage = channel?.firstMessage?.type === 'deleted';
       const isSelfChatChannel = channel?.type === 'messaging' && channel?.members?.length < 2;
-
-      return !isLocationChannel && !(isDeletedMessage || isSelfChatChannel);
+      const isCommunityChannel = channel?.type === 'topics';
+      const isDeletedCommunityMessage = channel?.messages[channel?.messages?.length - 1]?.text
+        ?.toLowerCase()
+        ?.includes('this message was deleted');
+
+      return (
+        !isLocationChannel &&
+        !(isDeletedMessage || isSelfChatChannel) &&
+        !((isDeletedMessage || isDeletedCommunityMessage) && isCommunityChannel)
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for filtering channels seems to be a bit complex and could be simplified. Also, the condition `isDeletedMessage || isDeletedCommunityMessage` is checked twice which is unnecessary. Here's a simplified version of your code:

```diff
-       const isCommunityChannel = channel?.type === 'topics';
-       const isDeletedCommunityMessage = channel?.messages[channel?.messages?.length - 1]?.text
-         ?.toLowerCase()
-         ?.includes('this message was deleted');
- 
-       return (
-         !isLocationChannel &&
-         !(isDeletedMessage || isSelfChatChannel) &&
-         !((isDeletedMessage || isDeletedCommunityMessage) && isCommunityChannel)
-       );
+       const isCommunityChannelWithDeletedMessage = channel?.type === 'topics' && channel?.messages[channel?.messages?.length - 1]?.text?.toLowerCase()?.includes('this message was deleted');
+
+       return !isLocationChannel && !isSelfChatChannel && !(isDeletedMessage && isCommunityChannelWithDeletedMessage);
```
This way, you only check once if the last message in a community channel was deleted, and the overall readability of the code is improved.


<!-- This is an auto-generated comment by OSS CodeRabbit -->